test/distributed/test_compute_comm_reordering.py

```diff
@@ -146,25 +164,43 @@ def func(a, *, tag, ranks, group_size):
     )
     def test_sink_waits_raise_comms(self):
         def func(a, *, tag, ranks, group_size):
-            c = torch.relu(a)
+            b = torch.matmul(a, a)
+            c = torch.relu(b)
             d = torch.matmul(c, c)
-            ar = _functional_collectives.all_reduce(a, "sum", ranks, tag)
-            e = d + ar
-            return (e,)
+            e = _functional_collectives.all_reduce(b, "sum", "0")
+            f = torch.relu(d)
+            g = torch.matmul(f, f)
+            return torch.mm(e, g)
 
         with _dynamo_dist_per_rank_init(self.rank, self.world_size):
             inputs = torch.ones(4, 4, dtype=torch.float, device="cuda") + self.rank
             compiled = torch.compile(func)
             code = run_and_get_triton_code(compiled, inputs, **self.get_world_trs())
-            # NOTE: notice that `dist.all_reduce` is raised above relu and matmul,
-            # and `_wait_tensor` is delayed until right before first use
-            FileCheck().check("dist.all_reduce(").check("triton_poi_fused_relu").check(
-                "_wait_tensor("
-            ).check("extern_kernels.addmm(").run(code)
+            # Things to verify:
+            # - The clone prologue of the all_reduce_ should not be fused with
+            # any relus.
+            # - The all_reduce_ and its prologue should be raised above the 2nd
+            # matmul but below the 1st matmul.
```

Comment:
Loved the ability to raise prologue!

I wonder should we add another unit test / enhance this unit test to show it can raise multiple prologue ops? e.g. 
```
b = torch.matmul(a, a)
c = torch.relu(b)
d = torch.matmul(c, c)
b1 = b + b
e = _functional_collectives.all_reduce(b1, "sum", "0")
```
the (`b1`, `e`) computation block should raise above the (`c`, `d`) computation block. 

Thanks!

Reply:
Added this enhancement in https://github.com/pytorch/pytorch/pull/130276/files#diff-5711354a54dd404142d5444e489df93d4a303bdc74b302ef64dfbb6d68b88e6aR170